console/views.py

```diff
@@ -65,3 +74,51 @@ def decrypt():
 
     else:
         return render_template('decrypt.html')
+
+
+def get_filtered_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest):
+    try:
+        q = db.session.query(SurveyResponse)
+        if tx_id != '':
+            q = q.filter(SurveyResponse.tx_id == tx_id)
+        if ru_ref != '':
+            q = q.filter(SurveyResponse.data["metadata"]["ru_ref"].astext == ru_ref)
+        if survey_id != '':
+            q = q.filter(SurveyResponse.data["survey_id"].astext == survey_id)
+        dt_column = func.date(SurveyResponse.data["submitted_at"].astext)
+        if datetime_earliest:
+            year = int(datetime_earliest[:4])
+            month = int(datetime_earliest[5:7])
+            day = int(datetime_earliest[8:10])
+            hour = int(datetime_earliest[11:13])
+            minute = int(datetime_earliest[14])
+            q = q.filter(dt_column > datetime(year, month, day, hour, minute))
+        if datetime_latest:
+            year = int(datetime_latest[:4])
+            month = int(datetime_latest[5:7])
+            day = int(datetime_latest[8:10])
+            hour = int(datetime_latest[11:13])
+            minute = int(datetime_latest[14])
+            q = q.filter(dt_column < datetime(year, month, day, hour, minute))
+        filtered_data = q.all()
+    except DataError as e:
+        logger.error("Invalid search term", error=e)
+        return []
+    except SQLAlchemyError as e:
+        logger.error("Database error", error=e)
+        return []
+
+    return filtered_data
+
+
+@app.route('/store', methods=['GET'])
+@flask_security.roles_required('SDX-Developer')
+def store():
+    tx_id = request.args.get('tx_id', type=str, default='')
```

Comment:
We need `tx_id=''` etc in `get_filtered_responses` because when a user clicks the search button with blank fields an empty string is submitted in the form.

The defaults here are a separate thing which I've also left as `=''` to be consistent with this.